### PR TITLE
update second review

### PR DESCRIPTION
**Pull request by rzmobiledev**

### CODE REVIEW

File: hangman.py [hunk 1]
```diff
@@ -1,76 +1,74 @@
...
-            if letters[i] == guess:
-                # add used letters to list
-                used_letters.append(guess)
+            for i in range(len(stored_letters)):
+                # accepts both lowercase and uppercase inputs
+                if (
+                    stored_letters[i] == guess
+                    or stored_letters[i] == guess.upper()
+                    or stored_letters[i] == guess.lower()
+                ):
```
Comment (josef-o) [inline hangman.py:59]: Good catch @jeelion22!

A way to do this in one line is using the same case on both sides:

```
stored_letters[i].upper() == guess.upper()
```

File: hangman.py [hunk 2]
```diff
@@ -1,76 +1,74 @@
...
 Used letters: j a                                                                                                                                                
 Word: j a _ a                                                                                                                                                    
 Guess a letter: v                                                                                                                                                
 You guessed the word java !
-'''
+"""
 
-count = 0
-count_balance = 6
+stored_letters = list("java")  # converts string to list
```
Comment (josef-o) [inline hangman.py:23]: This is the same as doing `["java"]` I'm not sure what you mean by *converts* but to be clear the result is not `["j", "a", "v", "a"]`

File: hangman.py [hunk 3]
```diff
@@ -1,76 +1,74 @@
...
+guessed_word = len(stored_letters) * ["_"]
+used_letters = list()
```
Comment (josef-o) [inline hangman.py:25]: Using `list()` is the same as doing `[]` which is a bit shorter if you like

**Review by jeelion22 — APPROVED** [review]
that's fine.

File: hangman.py [hunk 4]
```diff
@@ -1,76 +1,74 @@
...
 
-word = ["_", "_", "_", "_"]
-guessed_word = ["_", "_", "_", "_"]
 
-letters = ["j", "a", "v", "a"]
-used_letters = list()
+count_balance = 6
 
-while count < 6:
+while count_balance <= 6:
```
Comment (josef-o) [inline hangman.py:30]: `count_balance` is going down so just with this it becomes an infinite loop.

I know there is a `break` inside the loop but having both conditions is confusing as to which one matters.

File: hangman.py [hunk 5]
```diff
@@ -1,76 +1,74 @@
...
-            print("Only one char is allowed!")
+            print("Only one char is allowed")
             continue
 
-        if guess not in letters:
-            # start to count every loop
-            count += 1
-            # We start to minus 1 each count balance
+        # decrements the no.of tries if the guess is wrong
+        if guess not in stored_letters:
```
Comment (josef-o) [inline hangman.py:43]: There's a bug here, as it stands `guess` will never be in `stored_letters`

File: hangman.py [hunk 6]
```diff
@@ -1,76 +1,74 @@
...
+                    or stored_letters[i] == guess.upper()
+                    or stored_letters[i] == guess.lower()
+                ):
+                    guessed_word[i] = stored_letters[i]
 
-                guessed_word[i] = letters[i]
-                word[i] = letters[i]
-            
-        if guessed_word == letters:
+        if guessed_word == stored_letters:
```
Comment (josef-o) [inline hangman.py:62]: Bug here also.

**Review by josef-o — CHANGES_REQUESTED** [review]
Needs changes to be correct

File: hangman.py (hunk 2 above)
Comment (josef-o) [reply hangman.py:23]: Oops my bad I was wrong here

File: hangman.py (hunk 5 above)
Comment (josef-o) [reply hangman.py:43]: Disregard this I was wrong

File: hangman.py (hunk 6 above)
Comment (josef-o) [reply hangman.py:62]: Not

File: hangman.py (hunk 1 above)
Comment (rzmobiledev) [reply hangman.py:59]: @jeelion22 You can clone main branch first, and then in your terminal/bash from your current project, type `git checkout development` and enter then `git pull origin development`. You can start practice here. If you have any issues just let me know

**Comment by jeelion22** [comment]
I am reading your review.

**Comment by jeelion22** [comment]
okay

**Pull request closed without merging**